kafka-connector/src/main/java/com/google/pubsub/kafka/sink/CloudPubSubSinkTask.java

```diff
@@ -392,6 +401,11 @@ private void createPublisher() {
                     .setDelayThreshold(Duration.ofMillis(maxDelayThresholdMs))
                     .setElementCountThreshold(maxBufferSize)
                     .setRequestByteThreshold(maxBufferBytes)
+                    .setFlowControlSettings(FlowControlSettings.newBuilder()
```

Comment:
Can we only populate these if the maxOutstandingRequestBytes and maxOutstanding{ElementCount,Messages} is less than the default otherwise we'll unnecessarily slow down publishing by acquiring/releasing flow controller locks https://github.com/googleapis/java-pubsub/blob/0bba01661c01ca540625f63a0d7e89d43e2dab34/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java#L247-L257.

Reply:
Sure